narf/tfutils.py

```diff
@@ -1,47 +1,62 @@
 import tensorflow as tf
 
-def function_to_tflite(func, input_signature):
+def function_to_tflite(funcs, input_signatures):
     """Convert function to tflite model using python dynamic execution trickery to ensure that inputs
     and outputs are alphabetically ordered, since this is apparently the only way to prevent tflite from
     scrambling them"""
 
-    inputs = []
-    for i in range(len(input_signature)):
-        input_name = f"input_{i:05d}"
-        inputs.append(input_name)
+    if not isinstance(funcs, list):
+        funcs = [funcs]
+        input_signatures = [input_signatures]
 
-    def wrapped_func(*args):
-        outputs = func(*args)
+    def wrapped_func(iif, *args):
+        outputs = funcs[iif](*args)
 
         if not isinstance(outputs, tuple):
             outputs = (outputs,)
 
         output_dict = {}
         for i,output in enumerate(outputs):
-            output_name = f"output_{i:05d}"
+            output_name = f"output_{iif:05d}_{i:05d}"
             output_dict[output_name] = output
 
         return output_dict
 
-    arg_string = ", ".join(inputs)
+    arg_string = []
+    for iif, input_signature in enumerate(input_signatures):
+        inputs = []
+        if input_signature == None or input_signature == "":
```

Comment:
Can you explain what this special case is needed for?

Reply:
These are necessary in case there is no input argument to the TF function (e.g. for the meta info)